Neo4j.Driver/Neo4j.Driver/Internal/ConnectionValidator.cs

```diff
@@ -0,0 +1,116 @@
+using System;
+using System.Threading.Tasks;
+
+namespace Neo4j.Driver.Internal
+{
+    internal class ConnectionValidator
+    {
+        private readonly TimeSpan _connIdleTimeout;
+        private readonly TimeSpan _maxConnLifeTime;
+
+        public ConnectionValidator(TimeSpan connIdleTimeout, TimeSpan maxConnLifeTime)
+        {
+            _connIdleTimeout = connIdleTimeout;
+            _maxConnLifeTime = maxConnLifeTime;
+        }
+
+        public bool IsConnectionReusable(IPooledConnection connection)
```

Comment:
Shall we change this method's name to be more self describing such as ResetAndRestartIdleTimer because it has side effects aside from checks.